### PR TITLE
Better handling of different types in Elm generator

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.3.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

To fix https://github.com/OpenAPITools/openapi-generator/issues/1079

cc @trenneman

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java [hunk 1]
```diff
@@ -527,19 +529,26 @@ private String toOptionalValue(String value) {
 
     private String paramToString(final CodegenParameter param) {
         final String paramName = param.paramName;
-        if (param.isString || param.isUuid) {
+
+        if (param.isString || param.isUuid) { // string
             return paramName;
-        }
-        if (ElmVersion.ELM_018.equals(elmVersion)) {
+        } else if (ElmVersion.ELM_018.equals(elmVersion)) { // Elm 0.18
```
Comment (trenneman) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java]: All string-like types (for now also binary, file) should be in front of the Elm 0.18 logic.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java [hunk 2]
```diff
@@ -527,19 +529,26 @@ private String toOptionalValue(String value) {
...
-        }
-        if (param.isInteger || param.isLong) {
+        } else if (param.isInteger || param.isLong) { // int/long
             return "String.fromInt " + paramName;
-        }
-        if (param.isFloat || param.isDouble) {
+        } else if (param.isFloat || param.isDouble) { // float/double
             return "String.fromFloat " + paramName;
+        } else if (param.isBoolean) { // boolean
+            return "String.fromBool " + paramName;
```
Comment (trenneman) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java]: `String.fromBool` does not exists. Need to add a basic if expression here.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java [hunk 3]
```diff
@@ -527,19 +529,26 @@ private String toOptionalValue(String value) {
...
+        } else if (param.isDateTime || param.isDate) { // datetime
+            return paramName;
+        } else if (param.isBinary) { // binary
+            return paramName;
+        } else if (param.isByteArray) { // byte array
```
Comment (trenneman) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java]: What's with all the comments? Feels a bit ambiguous.

**Comment by trenneman** [comment]
My fix is ready. I had to rebase on #1104, so I'll wait for that one to get merged.

**Comment by wing328** [comment]
👌

**Comment by trenneman** [comment]
@wing328 I've updated the PR. Can you verify?

**Comment by wing328** [comment]
👍 Let's merge this PR so that we can include the enhancement in the 3.3.0 release.

**Pull request merged**